Core/Database/Repositories/EmpireRepository.cs

```diff
@@ -2,27 +2,67 @@
 using System.Data;
 using System.Threading.Tasks;
 using Dapper;
+using QuantumCore.Core.Cache;
 
 namespace QuantumCore.Database;
 
 public interface IEmpireRepository
 {
-    Task<byte?> GetEmpireForAccountAsync(Guid accountId);
+    Task<byte?> GetEmpireForPlayerAsync(Guid accountId);
+    Task<byte?> GetTempEmpireForAccountAsync(Guid accountId);
 }
 
 public class EmpireRepository : IEmpireRepository
 {
     private readonly IDbConnection _db;
+    private readonly ICacheManager _cacheManager;
 
-    public EmpireRepository(IDbConnection db)
+    public EmpireRepository(IDbConnection db, ICacheManager cacheManager)
     {
         _db = db;
+        _cacheManager = cacheManager;
     }
 
-    public async Task<byte?> GetEmpireForAccountAsync(Guid accountId)
+    public async Task<byte?> GetEmpireForPlayerAsync(Guid playerId)
     {
-        return await _db.QueryFirstOrDefaultAsync<byte>(
-            "SELECT Empire FROM account.accounts WHERE Id = @AccountId", new { AccountId = accountId });
+        var empireRedisKey = $"empire-pid:{playerId}";
+        var cachedEmpire = await _cacheManager.Get<byte?>(empireRedisKey);
+
+        if(cachedEmpire.HasValue)
+        {
+            // If found in Redis on Player, return the cached empire value
+            return cachedEmpire;
+        }
+
+        var databaseEmpire = await _db.QueryFirstOrDefaultAsync<byte>(
+            "SELECT Empire FROM game.players WHERE Id = @PlayerId", new { PlayerId = playerId });
+
+        if (databaseEmpire > 0)
+        {
+            await _cacheManager.Set(empireRedisKey, databaseEmpire);
+        }
+
+        return databaseEmpire;
+    }
+
+    /// <summary>
```

Comment:
Put this summary on the interface. We will only ever talk to the interface and see it's documentation. We (as consumer) will not see the implementation documentation